nightly_tests/smoke_test.sh

```diff
@@ -1,45 +1,91 @@
+fi
 
-# Lambda Function:
-# RandomStringLambdaFunction
+# Get all resource IDs from the CloudFormation stack
+resources=$(aws cloudformation list-stack-resources --stack-name "$STACK_NAME" --query "StackResourceSummaries[].[ResourceType,PhysicalResourceId]" --output text)
 
-# SSM Parameters:
-# UnityProjectName
-# UnityVenueName
+echo "Fetching tags for resources in stack '$STACK_NAME'..."
 
-# EC2 Launch Template:
-# ManagmentConsoleLaunchTemplate
+# Loop through resources and fetch tags
+while read -r resourceType resourceId; do
+    case $resourceType in
+        AWS::ElasticLoadBalancingV2::Listener|AWS::Lambda::Function|AWS::CloudFormation::CustomResource|AWS::SSM::Parameter)
+            # Skip fetching tags for these resources
+            continue
+            ;;
+    esac
 
-# AutoScaling Group:
-# DeployerAutoScalingGroup
+    echo "Resource: $resourceType, ID: $resourceId"
+    case $resourceType in
+        AWS::EC2::Instance|AWS::EC2::SecurityGroup|AWS::EC2::LaunchTemplate)
+            tags=$(aws ec2 describe-tags --filters "Name=resource-id,Values=$resourceId" --query "Tags[].[Key,Value]" --output text)
+            ;;
+        AWS::IAM::Role)
+            tags=$(aws iam list-role-tags --role-name "$resourceId" --query "Tags[].[Key,Value]" --output text)
+            ;;
+        AWS::IAM::InstanceProfile)
+            # Extract the role name from the instance profile
+            roleName=$(aws iam get-instance-profile --instance-profile-name "$resourceId" --query 'InstanceProfile.Roles[0].RoleName' --output text)
+            if [ "$roleName" != "None" ]; then
+                tags=$(aws iam list-role-tags --role-name "$roleName" --query "Tags[].[Key,Value]" --output text)
+            else
+                tags=""
+            fi
+            ;;
+        AWS::AutoScaling::AutoScalingGroup)
+            # AutoScaling Group names 
+            tags=$(aws autoscaling describe-tags --filters "Name=auto-scaling-group,Values=$resourceId" --query "Tags[].[Key,Value]" --output text)
+            ;;
+        AWS::ElasticLoadBalancingV2::LoadBalancer|AWS::ElasticLoadBalancingV2::TargetGroup)
+            # For ELBv2 (ALB/NLB)
+            tags=$(aws elbv2 describe-tags --resource-arns "$resourceId" --query "TagDescriptions[0].Tags[].[Key,Value]" --output text)
+            ;;
+        *)
+            echo "Skipping resource type $resourceType for tagging details."
+            tags=""
+            ;;
+    esac
+    if [ -n "$tags" ]; then
+        while read -r key value; do
+            echo "  Tag: $key, Value: $value"
+        done <<< "$tags"
+    else
+        echo "  No tags found or unable to retrieve tags for this resource type."
+    fi
+    echo # Newline 
+done <<< "$resources"
 
-# Security Groups:
-# ManagementConsoleSecurityGroup
-# ALBSecurityGroup
-
-# Elastic Load Balancing (ELB):
-# MCTargetGroup
-# ApplicationLoadBalancer
-# PublicLoadBalancerListener
 
 # List of SSM Parameter accessed by cloudformation stack
```

Comment:
This is really a combination of SSM parameters "accessed by" and "created by" the CF stack